### PR TITLE
Fix flaky isLaterThanDays date test

### PR DESCRIPTION
**Pull request by gpunto**
### Goal

`DateExtensionsTests` flakes on CI. The "exactly as old as specified days" case builds a date `now - 5 days` and expects `isLaterThanDays(5 days)` to be false, but `isLaterThanDays` reads the clock again internally, so it only holds when both reads land in the same millisecond. Any tick between them flips the result and fails the test.

Closes [AND-1514](https://linear.app/stream/issue/AND-1514/flaky-dateextensionstests-islaterthandays-boundary-test-races-the)

### Implementation

- Add a defaulted `now: Date = Date()` parameter to `isLaterThanDays` so the reference instant can be injected. Production callers use the default and are unchanged.
- Derive every test's date from a single fixed `now` and pass it in, removing the wall-clock dependence from the whole suite.

Both are `internal`, so no public API change.

### Testing

- `:stream-chat-android-client:testDebugUnitTest --tests "*DateExtensionsTests"` passes; the boundary case is now deterministic.

## Summary by CodeRabbit

* **Refactor**
  * Improved date comparison consistency by allowing comparisons to use a specified reference date.
  * Updated internal date tests to use shared, consistent time values.

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/internal/Date.kt [hunk 1]
```diff
@@ -19,9 +19,8 @@ package io.getstream.chat.android.client.extensions.internal
 import java.util.Date
 
 /**
- * Checks if the Date is later than [daysInMillis].
+ * Checks if the Date is older than [daysInMillis], measured against [now].
  */
-internal fun Date.isLaterThanDays(daysInMillis: Long): Boolean {
-    val now = Date()
+internal fun Date.isLaterThanDays(daysInMillis: Long, now: Date = Date()): Boolean {
```
Comment (andremion) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/internal/Date.kt]: Small thing: the KDoc now says "older than" but the name still says "later than". The function returns true when the receiver is further in the past, so the doc is the correct one. It is `internal` with a single caller (`ChatClient.checkSyncHistoryPreconditions`), so renaming it to `isOlderThanDays` would be cheap. Happy to leave it if you prefer to keep the diff small.

**Review by andremion — APPROVED** [review]
Looks good, the injected `now` makes the boundary case deterministic. One naming nit inline, optional.

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/internal/Date.kt (hunk 1 above)
Comment (gpunto) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/internal/Date.kt]: Good call, renamed to isOlderThanDays in 24f3112 (the function, its single caller in ChatClient, and the tests). Same rename on the develop port.
